### PR TITLE
Use major version numbers for actions.

### PR DESCRIPTION
**Pull request by ashleypittman**
Signed-off-by: Ashley Pittman <ashley@pittman.co.uk>

### CODE REVIEW

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -7,8 +7,7 @@ on:
   pull_request:
     branches: [master]
 
-permissions:
-  contents: read
+permissions: {}
```
Comment (bsbernd) [inline .github/workflows/codespell.yml:11]: The permission change needs to be documented in the commit message. I see that it disables all. This comes from https://github.com/libfuse/libfuse/issues/792 - what speaks against read permissions?

File: .github/workflows/codespell.yml [hunk 2]
```diff
@@ -17,6 +16,6 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        uses: actions/checkout@v4
```
Comment (bsbernd) [inline .github/workflows/codespell.yml:19]: See this commit please
```
commit 0d7f43f110ec2a246bda906d4190767f8626446c
Author: Pedro Kaj Kjellerup Nacht <pnacht@google.com>
Date:   Wed Jul 26 13:48:07 2023 +0000

    Hash-pin workflow Actions
    
    Signed-off-by: Pedro Kaj Kjellerup Nacht <pnacht@google.com>
```

And the reason is documented here: https://github.com/libfuse/libfuse/issues/819 (should have better been part of the commit message.

Also read this please
https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

So I think we better keep the current way.

**Review by bsbernd — CHANGES_REQUESTED** [review]

File: .github/workflows/codespell.yml (hunk 1 above)
Comment (ashleypittman) [reply .github/workflows/codespell.yml:11]: Whilst this check does read from the repository it does so anonymously without using the token so my takeaway was that the token could be granted with no permissions.

File: .github/workflows/codespell.yml (hunk 2 above)
Comment (ashleypittman) [reply .github/workflows/codespell.yml:19]: That makes a very good point.

**Pull request closed without merging**